### DISCUSSION TITLE
POLL: Lower maximum note size dimensions in osu!taiko.

### DISCUSSION THREAD

**Discussion by Raphalge** [ppy/osu#23025] [Polls]
This is a discussion that has been had for several years so while I may not be the first, I'm well overdue for taking it to GitHub.

Since around 2020 more and more people having taken to massively increasing the note sizes in their skins in order to gain several advantages with the HardRock mod, as well as making the combination Hidden+HardRock much more playable (This is however a different issue entirely.)

The HardRock mod in osu!taiko adds a multiplier to scroll-speed depending on your current window size. 
With wider resolutions, there's naturally more space for notes. But to compensate, they are also spaced much further apart thanks to the aforementioned scroll-speed multiplier. 

The question therefore goes as follows: "If the notes are _too_ far apart, why don't I just make the notes bigger to eliminate the space?" 

Due to perhaps unintended design, this is completely possible! And very easy to do! The default note size dimensions are 128x128, so by simply opening any image editing program and doubling that to the current limit of 256x256, you've successfully made highly-spaced notes twice as easy to read!

I propose lowering the note size limit further, as with the current dilemma, people are pushing limits that would otherwise be pipe dreams, and invalidating high scroll-speed reading as a skill.
This can be compared to previous issues like TaikoMania skins/storyboards, the Flashlight skinning exploits and even the old osu!standard issue of using animated skin files to emulate different approach rates. It has even been brought up by Taiko World Cup organizer Mangomizer as a twitter poll in which a large majority voted for fixing this issue: https://twitter.com/mangomizer/status/1226886195646521344

https://www.youtube.com/watch?v=mnHJMyF4740 Here is a video comparing a 5:4 NoMod view using 128x128 notes, to a 1920x950 view using notes twice as big with HardRock enabled. As you can hopefully see, they are largely indistinguishable and should paint a good picture of the issue.

**Comment by Raphalge** [comment]
I deeply apologize if this is not the place for stable discussions. (Seems like it is but I can't seem to move it to osu-stable-issues?)

**Comment by Garp0** [comment]
Hi, chiming in as an experienced Taiko player and Hard Rock player.
I'll come out of the gates swinging,
Everyone who has voted No has either zero understanding of what any of above means and has no place voting or is someone who is currently abusing said exploit. 

This is a straight up exploit and should be deemed no less. When in conjunction with Taiko's other blaring problem (Custom Resolutions which is a topic of it's own) it becomes amplified to such an extreme that the only thing Hard Rock does as a modification is increase OD.
Hard Rock essentially doubles base scroll speed and with these two game breaking exploits, it quite literally reads the same as if you didn't have the modification at all. 

In a Gamemode where the scroll speed plays a major part in the difficulty of some maps (or the entire basis of a mod which is used in Official Tournaments as an entire category). Being able to essentially change the AR equivalent through broken means on a gamemode with a fixed speed ruins the sportsmanship and competitiveness of it in it's entirety when you can just negate it full stop.
Even as we speak now, as TWC is running, I can tell you that certain teams that may be preforming exceptionally well on the faster Hard Rocks are making use of this exploit and it is giving them a completely unfair advantage to those who do not wish to Make a Custom Skin, Open their .osu file in the osu folder and adjust their resolution to a non-standard one mid match. 

I don't want to say this is as bad as TaikoMania but atleast taikomania players still had to deal with the challenges of reading while this one completely invalidates all that Hard Rock stands for. 
Thank you.

**Comment by peppy** [comment]
As part of skinning changes we are intending to enforce maximum sizing based on the template skin, which would include this case. There's no real discussion to be had because as mentioned, this is basically an exploit in stable that has been overlooked until now.

We will do our best to keep the maximum size large enough to allow the use cases where this is required for visual skin purposes (ie. showing the "don" / "kat" text below the standard location for objects on an authentic TnT skin)

**Comment by peppy** [comment]
As part of skinning changes we are intending to enforce maximum sizing based on the template skin, which would include this case. There's no real discussion to be had because as mentioned, this is basically an exploit in stable that has been overlooked until now.

We will do our best to keep the maximum size large enough to allow the use cases where this is required for visual skin purposes (ie. showing the "don" / "kat" text below the standard location for objects on an authentic TnT skin)

**Reply by Raphalge** [reply]
Bless your soul, thank you so much for acknowledging the issue. Since the main culprit is width this should be a great solution!

**Reply by Kasumisama** [reply]
Insanely good move.
I'm just wondering as well, as in osu!catch, you can essentially edit the note size to honestly whatever, can literally make CS8 look and feel like CS4 to hit just fine, and vice-versa for stuff from like CS2 to CS5 or whatever.
Would this count as another exploit or not really?

**Reply by peppy** [reply]
All gameplay elements will be limited in size in a pass at some point.

Let's track this at https://github.com/ppy/osu/issues/23099

**Reply by AetherRitual** [reply]
Will there be changes to minimum size for notes? Or will it only affect bignote skins? 
Or is it impossible to patch smallnote/nobig since you can just make the outer part transparent (or something like that).
